### Project 3 - Capstone Project (The Impact Covid-19 had on Medicaid)

#### Outline:
##### 1. Data Importing/Cleaning/Filtering
##### 2. Statistics
##### 3. Visualizations
##### 4. Conclusion

In [4]:
import pandas as pd
import numpy as np
from pivottablejs import pivot_ui
from IPython.display import HTML

In [5]:
df = pd.read_csv('Medicaid_Program_Enrollment_by_Month___Beginning_2009.csv', low_memory=False)

#### *1B. Getting to the know the data*

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2389919 entries, 0 to 2389918
Data columns (total 12 columns):
 #   Column                            Dtype 
---  ------                            ----- 
 0   Eligibility Year                  int64 
 1   Eligibility Month                 int64 
 2   Economic Region                   object
 3   Aid Category                      object
 4   Dual Eligible                     object
 5   Managed Care vs. Fee For Service  object
 6   Plan Name                         object
 7   Plan Type                         object
 8   Gender                            object
 9   Age Group                         object
 10  Race                              object
 11  Number of Recipients              int64 
dtypes: int64(3), object(9)
memory usage: 218.8+ MB


In [7]:
# Checking for nulls in the dataframe

df.isnull().sum().sum()

0

In [8]:
df.head()

Eligibility Year  Eligibility Month   Economic Region Aid Category  \
0              2009                  1  Capital District       FHPLUS   
1              2009                  1  Capital District       FHPLUS   
2              2009                  1  Capital District       FHPLUS   
3              2009                  1  Capital District       FHPLUS   
4              2009                  1  Capital District       FHPLUS   

  Dual Eligible Managed Care vs. Fee For Service Plan Name Plan Type  Gender  \
0            NO                              MMC     CDPHP  HMO/PHSP  Female   
1            NO                              MMC     CDPHP  HMO/PHSP  Female   
2            NO                              MMC     CDPHP  HMO/PHSP  Female   
3            NO                              MMC     CDPHP  HMO/PHSP  Female   
4            NO                              MMC     CDPHP  HMO/PHSP  Female   

  Age Group      Race  Number of Recipients  
0     00-20     ASIAN                     5  
1     00-20     BLACK                    13  
2     00-20  HISPANIC                    11  
3     00-20     OTHER                     3  
4     00-20     WHITE                    52

In [9]:
df['Eligibility Year'].value_counts()

2021    226578
2020    212071
2019    208557
2017    208267
2018    207322
2016    191349
2014    171842
2015    168766
2013    158889
2012    143505
2011    134539
2009    131987
2010    130041
2022     96206
Name: Eligibility Year, dtype: int64

In [10]:
df['Gender'].value_counts()

Female     1217748
Male       1171589
Unknown        582
Name: Gender, dtype: int64

In [11]:
df['Age Group'].value_counts()

45-64    767660
21-44    656432
65+      638247
00-20    327580
Name: Age Group, dtype: int64

In [12]:
df['Race'].value_counts()

WHITE              541734
BLACK              473033
HISPANIC           443591
OTHER              407878
ASIAN              336593
NATIVE AMERICAN    187090
Name: Race, dtype: int64

In [13]:
df['Plan Type'].value_counts()

HMO/PHSP              1143804
PARTIAL MLTC           503653
FEE-FOR-SERVICE        297853
HARP                   198933
MEDICAID ADVANTAGE      89287
PACE                    40856
MAP                     40166
AIDS SNP                38793
FIDA                    24309
PARTIAL CAP PLAN        10818
FFS PARTIAL CAP          1246
SENIOR HMO                201
Name: Plan Type, dtype: int64

In [37]:
# Filtering the data

years = [2018, 2019, 2020]
region = ['Long Island','New York City','Mid-Hudson']
filt = df['Eligibility Year'].isin(years) & (df['Economic Region'].isin(region)) & (df['Eligibility Month'] == 12) & (df['Plan Type'] == "HMO/PHSP") & (df['Gender'] != 'Unknown')
df2 = df.loc[filt]

In [38]:
df2['Eligibility Year'].value_counts()

2018    3097
2019    3074
2020    3050
Name: Eligibility Year, dtype: int64

In [39]:
df2['Gender'].value_counts()

Male      4652
Female    4569
Name: Gender, dtype: int64

In [40]:
df2['Plan Type'].value_counts()

HMO/PHSP    9221
Name: Plan Type, dtype: int64

In [41]:
df2['Economic Region'].value_counts()

New York City    3653
Mid-Hudson       2834
Long Island      2734
Name: Economic Region, dtype: int64

In [42]:
df2['Eligibility Month'].value_counts()

12    9221
Name: Eligibility Month, dtype: int64

#### *2. Statistics*

In [43]:
df2.groupby(['Economic Region','Eligibility Year'])['Number of Recipients'].sum()

Economic Region  Eligibility Year
Long Island      2018                 376377
                 2019                 372659
                 2020                 452930
Mid-Hudson       2018                 390859
                 2019                 389926
                 2020                 465716
New York City    2018                2503273
                 2019                2381947
                 2020                2782397
Name: Number of Recipients, dtype: int64

In [44]:
df2.groupby(['Eligibility Year'])['Number of Recipients'].mean()

Eligibility Year
2018    1056.024863
2019    1022.944697
2020    1213.456721
Name: Number of Recipients, dtype: float64

In [45]:
df2.groupby(['Race','Gender','Eligibility Year'])['Number of Recipients'].sum()

Race             Gender  Eligibility Year
ASIAN            Female  2018                194903
                         2019                183325
                         2020                203831
                 Male    2018                183146
                         2019                171538
                         2020                190542
BLACK            Female  2018                275031
                         2019                253656
                         2020                291689
                 Male    2018                229719
                         2019                213804
                         2020                251217
HISPANIC         Female  2018                210792
                         2019                191865
                         2020                204771
                 Male    2018                165307
                         2019                152528
                         2020                163405
NATIVE AMERICAN  Femal

In [46]:
df2.groupby(['Age Group','Eligibility Year'])['Number of Recipients'].sum()

Age Group  Eligibility Year
00-20      2018                1551775
           2019                1497166
           2020                1640447
21-44      2018                1012133
           2019                 975558
           2020                1247989
45-64      2018                 650512
           2019                 624161
           2020                 741145
65+        2018                  56089
           2019                  47647
           2020                  71462
Name: Number of Recipients, dtype: int64

In [47]:
# Pivot table
df_pivot = df2.pivot_table(index='Eligibility Year',values='Number of Recipients'
                          ,aggfunc='sum')
df_pivot

Number of Recipients
Eligibility Year                      
2018                           3270509
2019                           3144532
2020                           3701043

In [48]:
# Drop nulls and percent change
df3 = np.round(df_pivot.pct_change()*100,0)
df3 = df3.dropna()
print(df3)

                  Number of Recipients
Eligibility Year                      
2019                              -4.0
2020                              18.0


In [49]:
df_pivot2 = df2.pivot_table(index='Eligibility Year',columns='Gender',values='Number of Recipients'
                          ,aggfunc='sum')
df_pivot2

Gender             Female     Male
Eligibility Year                  
2018              1740390  1530119
2019              1670473  1474059
2020              1953468  1747575

In [50]:
df4 = np.round(df_pivot2.pct_change()*100,0)
df4 = df4.dropna()
print(df4)

Gender            Female  Male
Eligibility Year              
2019                -4.0  -4.0
2020                17.0  19.0


In [51]:
df_pivot3 = df2.pivot_table(index='Eligibility Year',columns='Race',values='Number of Recipients'
                          ,aggfunc='sum')
df_pivot3

Race               ASIAN   BLACK  HISPANIC  NATIVE AMERICAN    OTHER   WHITE
Eligibility Year                                                            
2018              378049  504750    376099             3311  1494914  513386
2019              354863  467460    344393             2957  1488931  485928
2020              394373  542906    368176             3337  1828832  563419

In [52]:
df5 = np.round(df_pivot3.pct_change()*100,0)
df5 = df5.dropna()
print(df5)

Race              ASIAN  BLACK  HISPANIC  NATIVE AMERICAN  OTHER  WHITE
Eligibility Year                                                       
2019               -6.0   -7.0      -8.0            -11.0   -0.0   -5.0
2020               11.0   16.0       7.0             13.0   23.0   16.0


In [53]:
df_pivot4 = df2.pivot_table(index='Eligibility Year',columns='Age Group',values='Number of Recipients'
                          ,aggfunc='sum')
df_pivot4

Age Group           00-20    21-44   45-64    65+
Eligibility Year                                 
2018              1551775  1012133  650512  56089
2019              1497166   975558  624161  47647
2020              1640447  1247989  741145  71462

In [54]:
df6=np.round(df_pivot4.pct_change()*100,0)
df6=df6.dropna()
print(df6)

Age Group         00-20  21-44  45-64   65+
Eligibility Year                           
2019               -4.0   -4.0   -4.0 -15.0
2020               10.0   28.0   19.0  50.0


#### *3. Visualizations*

In [55]:
from pivottablejs import pivot_ui

pivot_ui(df2)

In [56]:
from IPython.display import HTML

pivot_ui(df2, outfile_path='pivottable_2.html')

In [57]:
pivot_ui(df2, outfile_path='pivottable_3.html')

In [58]:
pivot_ui(df2, outfile_path='pivottable_4.html')

#### *4. Conclusion*

In summary, the COVID-19 pandemic resulted in an increase in unemployment,
which resulted with more people being eligible for Medicaid. In the New York City, 
Long Island and Mid-Hudson area saw a total increase of 557K (18%) Medicaid enrollees
in 2020. This was driven by an increase of 274K (19%) Male enrollees, 272K (28%) 
enrollees in the age group of 21-44 and 340K (23%) “Other” race.
